compiler/src/test/scala/org/scalajs/core/compiler/test/ScalaJSDefinedTest.scala

```diff
@@ -691,4 +691,39 @@ class ScalaJSDefinedTest extends DirectTest with TestHelpers {
     """
   }
 
+  @Test
+  def overrideJavaLangObjectMethod: Unit = {
+    val objs = List("class", "object")
```

Comment:
We should also have a test for what we want to happen in a `trait` (with abstract methods, obviously). We probably also want to warn in that case.

Reply:
We should also decide whether we want to warn for such definitions in `@js.native` stuff.